prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/git/GitClone.java

```diff
@@ -0,0 +1,80 @@
+package com.redhat.parodos.tasks.git;
+
+import com.google.common.io.Files;
+import com.redhat.parodos.tasks.jira.JiraTask;
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+
+import static com.redhat.parodos.workflow.context.WorkContextDelegate.getOptionalValueFromRequestParams;
+import static com.redhat.parodos.workflow.context.WorkContextDelegate.getRequiredValueFromRequestParams;
+
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import java.io.File;
+import java.util.List;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.InvalidRemoteException;
+import org.eclipse.jgit.api.errors.TransportException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@AllArgsConstructor
+public class GitClone extends BaseWorkFlowTask {
+
+	@Autowired
+	private GitUtils gitUtils;
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(
+				WorkParameter.builder().key(this.gitUtils.getUri()).type(WorkParameterType.TEXT).optional(false)
+						.description("Url to clone from").build(),
+				WorkParameter.builder().key(this.gitUtils.getBranch()).type(WorkParameterType.TEXT).optional(true)
+						.description("Branch to clone from, default main").build(),
+				WorkParameter.builder().key("credentials").type(WorkParameterType.TEXT).optional(false)
+						.description("Git credential").build());
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String gitUri = "";
+		String destination = "";
+		String gitBranch = "";
+
+		try {
+			gitUri = getRequiredValueFromRequestParams(workContext, this.gitUtils.getUri());
+			gitBranch = getOptionalValueFromRequestParams(workContext, this.gitUtils.getBranch(), "main");
+		}
+		catch (MissingParameterException e) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, e);
+		}
+
+		try {
+			destination = this.cloneRepo(gitUri, gitBranch);
+		}
+		catch (Exception e) {
```

Comment:
We could combine try block and have single catch block with `MissingParameterException | InvalidRemoteException | TransportException | GitAPIException`